service/src/main/java/bio/terra/pipelines/common/utils/StairwayFailedMetricsCounterHook.java

```diff
@@ -0,0 +1,47 @@
+package bio.terra.pipelines.common.utils;
+
+import static bio.terra.pipelines.common.utils.FlightUtils.inputParametersContainTrue;
+import static bio.terra.pipelines.common.utils.FlightUtils.isContextInvalid;
+
+import bio.terra.pipelines.app.common.MetricsUtils;
+import bio.terra.pipelines.dependencies.stairway.JobMapKeys;
+import bio.terra.pipelines.stairway.imputation.RunImputationJobFlightMapKeys;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.FlightStatus;
+import bio.terra.stairway.HookAction;
+import bio.terra.stairway.StairwayHook;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+/**
+ * A {@link StairwayHook} that logs a pipeline failure to Prometheus metrics upon flight failure.
```

Comment:
can we add some more documentation around what key to use to enable this functionality on a flight